File: 200-lambda.rb
ZERO = ->(f) { ->(v) {       v   } }
ONE  = ->(f) { ->(v) {    f.(v)  } }
TWO  = ->(f) { ->(v) { f.(f.(v)) } }

def to_integer(num)
  num.(->(x) { x + 1}).(0)
end

puts to_integer(TWO)

# next - subtraction.......... UGH....

File: js/200-lambda.js
const ZERO = (f) => { return (v) => { return     v   }}
const ONE  = (f) => { return (v) => { return   f(v)  }}
const TWO  = (f) => { return (v) => { return f(f(v)) }}


function toInt(x) {
  return x((v) => { return v + 1 })(0)
}

console.log(toInt(ZERO))
console.log(toInt(ONE))
console.log(toInt(TWO))

// next - subtraction.......... UGH....
